src/state-schemas/state-schema.js

```diff
@@ -182,6 +186,38 @@ export const stateProperties = {
     ],
     default: "KERNEL_IDLE"
   },
+  notebookHistory: {
+    type: "object",
+    properties: {
+      errorGettingRevisionContent: {
+        type: "boolean",
+        default: undefined
+      },
+      errorGettingRevisionList: {
+        type: "boolean",
+        default: undefined
+      },
+      revisionList: {
+        type: "array",
+        items: {
+          type: "object",
+          properties: {
+            id: { type: "integer" },
+            title: { type: "string" },
+            created: { type: "string" }
+          }
+        },
+        default: []
+      },
+      selectedRevision: {
+        type: "number"
+      },
+      revisionContent: {
```

Comment:
I'm seperating out the fetching of the revision metadata + their content because the sum of every revision + its content might be prohibitively large in certain cases. I can do what you suggest though wrt additionalProperties.

Reply:
right, but you can start with the `content: undefined`, and then lazily add it when you fetch the revision -- it's just a matter of where you put the data. putting the actual content within the rest of the revisionList seems cleaner overall.

Reply:
Storing the content within a separate object simplifies look-up when it comes time to actually retrieve the content for display. It's also easier to just blow away the revisionList structure when we want to download/refresh a new one, since nothing major will be lost. I'd rather leave this part as-is.

Reply:
gotcha, sounds good :+1:

Reply:
oops, i think the comment about adding `additionalProperties: {type: "string"}` to the schema for `revisionContent` got lost in the thread